### ISSUE TITLE
Modal doesn't close when clicking overlay

### ISSUE THREAD

**Issue by mmellado** [airtame/airtame-gooey#26]
We need to find a way to allow for the modal to close when clicking the overlay, while disallowing this behavior for the Dialog.

I'm thinking of doing this by having a new prop on the `Modal` component which would default to "close on overlay click" to true, but can be disabled through a prop. Something along the lines of

```jsx
<Modal
  isOpen={...}
  onClose={...}
  disableOverlayClose
>
....
</Modal>
```

**Comment by mmellado** [comment]
After internal discussion, we'll go with the behavior mentioned above

**Comment by mmellado** [comment]
Fixed by #27
